### ISSUE TITLE
DL3018 triggered on trailing escape newline

### ISSUE THREAD

**Issue by dhoepelman** [hadolint/hadolint#1031]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

hadolint treats an escape character as the last line the same as dockerfile

### Actual behavior

```
DL3018 warning: Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
```

### Steps to reproduce the behavior

**Output of `hadolint --version`

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** 

```Dockerfile
RUN apk add --no-cache \
    foo=1.0 \
    bar=2.0 \

# More commands here
```
